### PR TITLE
Replace Object.assign with Object destructuring (...)

### PR DESCRIPTION
**Pull request by lucasrmendonca**
### Description, Motivation and Context
- Object destructuring can be done conditionally, which makes us able to get rid of some if/else logic
- Also it looks even cleaner

### What kind of change does this PR introduce?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactoring (no new functionality, only code improvements)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] Overall test coverage is not decreased.
- [x] All new and existing tests passed.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/wampy.js [hunk 1]
```diff
@@ -379,12 +379,11 @@ class Wampy {
...
-            this._options = Object.assign({}, this._options, options);
-        } else if (this._isPlainObject(url)) {
-            this._options = Object.assign({}, this._options, url);
-        }
-
+        this._options = {
+            ...this._options,
+            ...(this._isPlainObject(options) ? options : {}),
+            ...(this._isPlainObject(url) ? url : {}),
+        };
```
Comment (KSDaemon) [inline src/wampy.js:386]: Well... I personally find the original version cleaner. In this case, we have 2 independent checks while they are mutually exclusive. It's not a problem here. But we even do not save LoCs...

What do you think?

Comment (lucasrmendonca) [reply src/wampy.js:386]: What if we kept the if/else block but changed only the `Object.assign()` part to the `...` syntax?

Comment (KSDaemon) [reply src/wampy.js:386]: Agree! Compromise! Let's make it :)

**Comment by KSDaemon** [comment]
👍🏻

**Comment by KSDaemon** [comment]
What's up with tests?

**Comment by lucasrmendonca** [comment]
> What's up with tests?

I typed `options` instead of `_options` ... fixing it right now, sorry :P

**Pull request merged**